internal/service/users/user.go

```diff
@@ -125,3 +125,20 @@ func (s *Service) findUser(user *types.User) (*types.User, error) {
 	}
 	return user, types.NewErrServerError(result.Error)
 }
+
+func (s *Service) SearchEntraForUsersAndMatch(ctx context.Context, query string) ([]openapi.UserData, error) {
+	// query entra
+	usernames, err := s.entra.FindUsernames(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	// then match to users in our db
+	users := []types.User{}
```

Comment:
How come we need to search entra first?

Reply:
we don't store the email and sometimes the userprincipalname is in the EXT format, so we'll have to search entra anyway for each query, it just seemed to make more sense to search entra and then match it to our records. The original/alt way would've been searching our db, if nothing then searching entra anyway and then matching to our db.
Also users _shouldn't_ exist on the portal without existing in the tenant (laura@example is the exception)

Reply:
I'm pro search only! but will need some user feedback before we decide

Reply:
but also, presently I think entra does actually [limit](https://github.com/ucl-arc-tre/portal/blob/4e517bcc1c8631ffa7957399796d1f6b1a2599a8/internal/controller/entra/main.go#L202C5-L202C5) the page length to 20, we're just not utilising it on the frontend

Reply:
Yeah, that all makes sense. Happy to go with this for now and refine further once we get feedback 👍